38_EXRSplit/main.cpp

```diff
@@ -1,94 +1,89 @@
-// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
 // This file is part of the "Nabla Engine".
 // For conditions of distribution and use, see copyright notice in nabla.h
-#include "nabla.h"
 
 
-#include <iostream>
-#include <cstdio>
-
+#include "../common/MonoSystemMonoLoggerApplication.hpp"
 
 using namespace nbl;
 using namespace core;
 using namespace asset;
 using namespace system;
 
 
-int main(int argc, char * argv[])
+// instead of defining our own `int main()` we derive from `nbl::system::IApplicationFramework` to play "nice" wil all platofmrs
+class HelloComputeApp final : public nbl::examples::MonoSystemMonoLoggerApplication
 {
-	// need to call this to Delay-Load DLLs properly
-	IApplicationFramework::GlobalsInit();
+	using base_t = examples::MonoSystemMonoLoggerApplication;
+public:
+	// Generally speaking because certain platforms delay initialization from main object construction you should just forward and not do anything in the ctor
+	using base_t::base_t;
 
-	smart_refctd_ptr<ISystem> system = nullptr;
+	// we stuff all our work here because its a "single shot" app
+	bool onAppInitialized(smart_refctd_ptr<ISystem>&& system) override
 	{
-	#ifdef _NBL_PLATFORM_WINDOWS_
-		system = make_smart_refctd_ptr<nbl::system::CSystemWin32>();
-	#elif defined(_NBL_PLATFORM_LINUX_)
-		system = make_smart_refctd_ptr<nbl::system::CSystemLinux>();
-	#else
-	#error "Unsupported Platform"
-	#endif
-	}
-
-	#if defined(_NBL_PLATFORM_WINDOWS_)
-	auto logger = core::make_smart_refctd_ptr<system::CColoredStdoutLoggerWin32>();
-	#else
-	auto logger = core::make_smart_refctd_ptr<system::CColoredStdoutLoggerANSI>();
-	#endif
-	auto assetManager = core::make_smart_refctd_ptr<nbl::asset::IAssetManager>(nbl::core::smart_refctd_ptr(system));
-
-	const bool isItDefaultImage = argc == 1;
-	if (isItDefaultImage)
-		logger->log("No image specified - loading a default OpenEXR image placed in media/OpenEXR!", ILogger::ELL_INFO);
-	else if (argc == 2)
-		logger->log((argv[1] + std::string(" specified!")).c_str(), ILogger::ELL_INFO);
-	else
-	{
-		logger->log("To many arguments - pass a single filename without .exr extension of OpenEXR image placed in media/OpenEXR!", ILogger::ELL_ERROR);
-		return 0;
-	}
```

Comment:
note that the old example was also a utility to split multi-channel EXRs, please restore the behaviour of using argv[1] correctly